content/concepts/content-management/031-fields.md

```diff
@@ -0,0 +1,103 @@
+    <FieldIndex>101</FieldIndex>
+    <ControlHint>sn:CustomControl</ControlHint>
+  </Configuration>
+</Field>
+```
+&nbsp;
+&nbsp;
+
+| | |
+|-|-|
+| name | Name of the field. Required, should be unique in a content type. |
+| type | Short name of the field type. This or handler is required. (see the list of field types above) |
+| handler | Fully qualified type name of field handler class. This or handler is required. |
+| DisplayName | Displayed name of the field. |
+| Description | Description of the field. |
+| Bind | Name of the storage property the field is bound to. |
+| Indexing | Indexing settings of the field. |
+| ReadOnly | Indicates if the field is read-only. |
+| Compulsory | Indicates if the field is required to fill. |
+| OutputMethod | Defines the field output type (Default, Raw, Text, Html) |
+| DefaultValue | Default value of the field. |
+| VisibleBrowse | Indicates if the field is visible in autogenerated browse view. |
+| VisibleEdit | Indicates if the field is visible in autogenerated edit view (while a content is edited). |
+| VisibleNew | Indicates if the field is visible in autogenerated new view (while a content is created). |
+| FieldIndex | Defines order of field in autogenerated views. |
+| ControlHint | Default field control name. Used by the mapper in control libs like [sn-controls-react](https://github.com/SenseNet/sn-client/tree/develop/packages/sn-controls-react). |
+
+## Search in fields
+
+For every content the field values can be indexed so that when searched for a value the corresponding content will appear in the result set. It is also possible to search in fields by explicitely defining the them. The way a specific field of a content is indexed is defined in the [field definition](/concepts/basics/04-search#metadataindexing,searchingbyfields).
+
+## Validation
+
+When the content is saved the input field data falls under validation according to the field settings. This includes non-empty value check for required fields and some field setting-dependent checks. An *Integer* field for example validates input data against data-type check - the entered value should be an integer value - and also checks whether input data falls between the range of *MinValue* and *MaxValue* as given in the field definition in the content type's CTD.
+
+## Fields on the ui
+
+Each repository has a content model, a schema that represents the content types you create. This schema is fully available to the client through the REST API. Using this schema it is possible to autogenerate the ui sinceit contains all the necessary information (field type, validations, etc). Can be a good starting point for this to check the [sn-controls-react](https://github.com/SenseNet/sn-client/tree/develop/packages/sn-controls-react) client library, that contains field controls for all the built-in field types and a mapper that manage how (with which field control) the specific field types should be displayed on the ui.
```

Comment:
```suggestion
Each repository has a content model, a schema that represents the content types you create. This schema is fully available to the client through the REST API. Using this schema it is possible to autogenerate the ui since it contains all the necessary information (field type, validations, etc). Can be a good starting point for this to check the [sn-controls-react](https://github.com/SenseNet/sn-client/tree/develop/packages/sn-controls-react) client library that contains field controls for all the built-in field types and a mapper that manage how (with which field control) the specific field types should be displayed on the ui.
```

Reply:
" since it contains" not "sinceit"